graphql_compiler/fast_introspection.py

```diff
@@ -0,0 +1,162 @@
+"""Module to execute GraphQL introspection query faster than graphql-core.
+
+On a large schema with tens of thousands of types, the approach using graphql-core's
+graphql_sync takes 26 seconds to run, whereas this module executes the same query in
+1.37 seconds. This module calls the same Type resolvers as graphql_sync to prevent
+code duplication with graphql-core.
+
+This module is fully dependent on the introspection query returned by `get_introspection_query`
+from graphql-core (with default parameters). Changes to that query will need to be
+reflected in this module. This module also assumes that the introspection query can
+be executed on the given schema.
+"""
+
+
+from typing import Any, Dict, Tuple, cast
+
+from graphql import (
+    GraphQLAbstractType,
+    GraphQLArgument,
+    GraphQLDirective,
+    GraphQLEnumType,
+    GraphQLEnumValue,
+    GraphQLField,
+    GraphQLNamedType,
+    GraphQLObjectType,
+    GraphQLSchema,
+    GraphQLType,
+    introspection_types,
+    is_abstract_type,
+)
+
+
+__Schema = cast(GraphQLObjectType, introspection_types["__Schema"])
+__Directive = cast(GraphQLObjectType, introspection_types["__Directive"])
+__DirectiveLocation = cast(GraphQLEnumType, introspection_types["__DirectiveLocation"])
+__Type = cast(GraphQLObjectType, introspection_types["__Type"])
+__Field = cast(GraphQLObjectType, introspection_types["__Field"])
+__InputValue = cast(GraphQLObjectType, introspection_types["__InputValue"])
+__EnumValue = cast(GraphQLObjectType, introspection_types["__EnumValue"])
+__TypeKind = cast(GraphQLEnumType, introspection_types["__TypeKind"])
+
+
+def _get_type_ref(type_: GraphQLType) -> Dict[str, Any]:
```

Comment:
All these functions need docstrings that explain their purpose, otherwise this code is going to be very difficult to read and maintain.

cc @bojanserafimov @pmantica1 any idea why pydocstyle doesn't seem to mind the fact that docstrings are missing?

Reply:
Seems like it's not configured to. http://www.pydocstyle.org/en/2.1.1/error_codes.html D102, D103 are missing.

Interesting that our `test_.*\.py` pydocstyle config has D102, therefore it's more strict than the non-test one.

Reply:
The config seems correct. Another point of weirdness that might help: when I run pydocstyle locally as part of the lint script, it always fails.

Reply:
Looking into it ..

Reply:
http://www.pydocstyle.org/en/2.1.1/error_codes.html It seems like pydocstyle doesn't look at private functions